hazelcast/src/main/java/com/hazelcast/map/impl/operation/TriggerLoadIfNeededOperation.java

```diff
@@ -33,7 +36,7 @@ public TriggerLoadIfNeededOperation(String name) {
 
     @Override
     public void run() {
-        isLoaded = recordStore.isLoaded();
+        isLoaded = recordStore.isKeyLoadFinished();
```

Comment:
So if I understand correctly, previously this could report `true` if all `loadingFutures` were done and now it is only reported if the `keyLoadFinished` is set to true? 
And the problem was that the futures were not migrated property but the result of the `keyLoadFinished` is migrated properly?

Reply:
Ok I am a bit closer to understanding the change - the `isKeyLoadFinished` will report if keys have been loaded (the values might not be loaded yet) and the `isLoaded` method reported if both keys and values have been loaded. I'm just missing how the previous behaviour might lead to the issue. Can you describe how the issue happens? I believe it has something to do with how the futures and the loading status is migrated?

Reply:
yep, so the loadingFutures list is not migrated, which means when the partition lands on another node, the list is empty, but the problem is that the keyLoadFinished future is not migrated, so it lead to the map being stuck in not-loaded state.

This fix is a hack - it ignores the fact that there still may be futures in progress, but it will properly set the loaded state. The issue here is that the map may be set to loaded a bit earlier that it has actually been loaded (due to the futures scheduled from the node that was the owner of the partition previously), but this we won't address for now. It's a super complex fix, since we found some other issues related to it.